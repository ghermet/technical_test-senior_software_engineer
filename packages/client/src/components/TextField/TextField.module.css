.input::-webkit-outer-spin-button,
.input::-webkit-inner-spin-button {
  @apply m-0;
  -webkit-appearance: none;
}
