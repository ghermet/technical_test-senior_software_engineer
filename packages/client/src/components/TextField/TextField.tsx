import React from 'react'
import { MagnifyingGlassLeft } from '@pretto/picto'
import { ctwMerge } from '../../utility/ctw-merge'
import styles from './TextField.module.css'

interface TextFieldProps
  extends React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement> {}

const TextField = ({ className, ...props }: TextFieldProps) => (
  <fieldset className={ctwMerge('items-center inline-flex', className)}>
    <input
      {...props}
      className={ctwMerge(
        styles.input,
        'input-book-16 appearance-none bg-neutral4 border-2 border-neutral3 text-neutral1 outline-offset-2 h-16 -mr-10 pr-14 p-4 rounded-lg border-solid w-full placeholder-neutral2 transition-colors duration-150 ease-in-out',
        'focus:border-primary2 focus:outline-primary2 focus:outline-2 focus:outline-dashed',
        'focus-within:outline-primary2 focus-within:outline-2 focus-within:outline-dashed'
      )}
    />
    <MagnifyingGlassLeft />
  </fieldset>
)

export default TextField
