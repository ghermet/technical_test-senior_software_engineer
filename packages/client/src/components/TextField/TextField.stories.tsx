import React from 'react'
import TextField from './TextField'
import '../../styles/globals.generated.css'

export default {
  title: 'Components/TextField',
  component: TextField,
}

export const Default = () => <TextField className="w-80" placeholder='Search...' defaultValue="" />
