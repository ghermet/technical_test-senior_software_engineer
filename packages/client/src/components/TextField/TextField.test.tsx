import React from 'react'
import { render } from '@testing-library/react'
import TextField from './TextField'

describe('TextField', () => {
  it('should render a text field element with the correct default classnames and props', () => {
    const { container } = render(<TextField className="some-class" placeholder="Search..." defaultValue="" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
