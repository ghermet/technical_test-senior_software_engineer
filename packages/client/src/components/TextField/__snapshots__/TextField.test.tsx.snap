// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should render a text field element with the correct default classnames and props 1`] = `
<fieldset
  class="items-center inline-flex some-class"
>
  <input
    class="input input-book-16 appearance-none bg-neutral4 border-2 border-neutral3 text-neutral1 outline-offset-2 h-16 -mr-10 p-4 rounded-lg border-solid w-full placeholder-neutral2 transition-colors duration-150 ease-in-out focus:border-primary2 focus:outline-primary2 focus:outline-2 focus:outline-dashed focus-within:outline-primary2 focus-within:outline-2 focus-within:outline-dashed"
    placeholder="Search..."
    value=""
  />
  <svg
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.8,16.6A9.26,9.26,0,0,1,9.1,19a9.45,9.45,0,0,1-6.5-3A9.21,9.21,0,0,1,0,9.4,9.24,9.24,0,0,1,2.8,2.8,9.07,9.07,0,0,1,9.4,0a9.71,9.71,0,0,1,6.7,2.6,9.43,9.43,0,0,1,.6,13.2l7.4,7.4-.8.8ZM14.2,2.4A8.63,8.63,0,0,0,9.5,1,8.54,8.54,0,0,0,1,9.5a8.63,8.63,0,0,0,1.4,4.7,8.21,8.21,0,0,0,3.8,3.1,9.2,9.2,0,0,0,4.9.5,8.54,8.54,0,0,0,4.4-2.3,8.18,8.18,0,0,0,2.3-4.4,8.23,8.23,0,0,0-.5-4.9A8.74,8.74,0,0,0,14.2,2.4Z"
    />
  </svg>
</fieldset>
`;
